examples/localclient_linux/route/main.go

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) 2021 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	linux_l3 "go.ligato.io/vpp-agent/v3/proto/ligato/linux/l3"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/namsral/flag"
+	"go.ligato.io/cn-infra/v2/agent"
+	"go.ligato.io/cn-infra/v2/logging"
+	"go.ligato.io/cn-infra/v2/logging/logrus"
+
+	"go.ligato.io/vpp-agent/v3/clientv2/linux/localclient"
+	"go.ligato.io/vpp-agent/v3/cmd/vpp-agent/app"
+	linux_ifplugin "go.ligato.io/vpp-agent/v3/plugins/linux/ifplugin"
+	linux_nsplugin "go.ligato.io/vpp-agent/v3/plugins/linux/nsplugin"
+	"go.ligato.io/vpp-agent/v3/plugins/orchestrator"
+	vpp_ifplugin "go.ligato.io/vpp-agent/v3/plugins/vpp/ifplugin"
+	linux_intf "go.ligato.io/vpp-agent/v3/proto/ligato/linux/interfaces"
+	vpp_intf "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/interfaces"
+)
+
+var (
+	timeout = flag.Int("timeout", 5, "Timeout between applying of initial and modified configuration in seconds")
+)
+
+/* Vpp-agent Init and Close*/
+
+// Start Agent plugins selected for this example.
+func main() {
+	// Set inter-dependency between VPP & Linux plugins
+	vpp_ifplugin.DefaultPlugin.LinuxIfPlugin = &linux_ifplugin.DefaultPlugin
+	vpp_ifplugin.DefaultPlugin.NsPlugin = &linux_nsplugin.DefaultPlugin
+	linux_ifplugin.DefaultPlugin.VppIfPlugin = &vpp_ifplugin.DefaultPlugin
+
+	// Init close channel to stop the example.
+	exampleFinished := make(chan struct{})
+
+	// Inject dependencies to example plugin
+	ep := &RouteExamplePlugin{
+		Log:          logging.DefaultLogger,
+		VPP:          app.DefaultVPP(),
+		Linux:        app.DefaultLinux(),
+		Orchestrator: &orchestrator.DefaultPlugin,
+	}
+
+	// Start Agent
+	a := agent.NewAgent(
+		agent.AllPlugins(ep),
+		agent.QuitOnClose(exampleFinished),
+	)
+	if err := a.Run(); err != nil {
+		log.Fatal(err)
+	}
+
+	go closeExample("VRF localhost example finished", exampleFinished)
+}
+
+// Stop the agent with desired info message.
+func closeExample(message string, exampleFinished chan struct{}) {
+	time.Sleep(time.Duration(*timeout+5) * time.Second)
```

Comment:
`*timeout+5` should be replaced with e.g. `*timeout*3` (there is multiplication so it scales when different timeout is specified at CLI; and the overall duration is longer so the agent does not stop as soon as the last transaction is sent, possibly before receiving reply).

Suggestion: `RouteExamplePlugin` would signal (by sending message through channel) when it finishes its job at line 130 and function `closeExample` would use this signal instead of using timeout. This would make it less error-prone is someone changes/expands the `AfterInit` method.